diff --git a/model/user.go b/model/user.go
index 08ffabc..ff26437 100644
--- a/model/user.go
+++ b/model/user.go
@@ -16,16 +16,16 @@ import (
 	"gorm.io/gorm"
 )
 
-const UserNameMaxLength = 20
+const UserNameMaxLength = 40
 
 // User if you add sensitive fields, don't forget to clean them in setupLogin function.
 // Otherwise, the sensitive information will be saved on local storage in plain text!
 type User struct {
 	Id               int                        `json:"id"`
-	Username         string                     `json:"username" gorm:"unique;index" validate:"max=20"`
+	Username         string                     `json:"username" gorm:"unique;index" validate:"max=40"`
 	Password         string                     `json:"password" gorm:"not null;" validate:"min=8,max=20"`
 	OriginalPassword string                     `json:"original_password" gorm:"-:all"` // this field is only for Password change verification, don't save it to database!
-	DisplayName      string                     `json:"display_name" gorm:"index" validate:"max=20"`
+	DisplayName      string                     `json:"display_name" gorm:"index" validate:"max=40"`
 	Role             int                        `json:"role" gorm:"type:int;default:1"`   // admin, common
 	Status           int                        `json:"status" gorm:"type:int;default:1"` // enabled, disabled
 	Email            string                     `json:"email" gorm:"index" validate:"max=50"`
